[BUG] hive_parquet_write_test.py: test_write_compressed_parquet_into_hive_table integration test failures

**Describe the bug**
Integration test failures seen in latest nightly builds in Spark-3.5.1
test_write_compressed_parquet_into_hive_table

**Reason:**
```
pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
```

**Failures:**
```
[2024-06-01T00:46:12.165Z] - generated xml file: /home/jenkins/agent/workspace/jenkins-rapids_integration-dev-github-1001-3.5.1/jars/integration_tests/target/run_dir-20240531223155-oYse/TEST-pytest-1717194715820773646.xml -

[2024-06-01T00:46:12.165Z] =========================== short test summary info ============================

[2024-06-01T00:46:12.165Z] FAILED ../../src/main/python/hive_parquet_write_test.py::test_write_parquet_into_hive_table[[Map(String(not_null),String), Map(Boolean(not_null),Boolean), Map(Byte(not_null),Byte), Map(Short(not_null),Short), Map(Integer(not_null),Integer), Map(Long(not_null),Long), Map(Float(not_null),Float), Map(Double(not_null),Double), Map(Timestamp(not_null),Timestamp), Map(Date(not_null),Date), Map(Decimal(not_null)(19,1),Decimal(19,1)), Map(Decimal(not_null)(36,5),Decimal(36,5))]-CTTW][DATAGEN_SEED=1717194715, TZ=UTC, IGNORE_ORDER({'local': True})] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.165Z] Project [map_from_arrays(transform(map_keys(_c0#298266), lambdafunction(lambda key#298398, lambda key#298398, false)), transform(map_values(_c0#298266), lambdafunction(lambda value#298400, lambda value#298400, false))) AS _c0#298402, map_from_arrays(transform(map_keys(_c1#298267), lambdafunction(lambda key#298403, lambda key#298403, false)), transform(map_values(_c1#298267), lambdafunction(lambda value#298405, lambda value#298405, false))) AS _c1#298407, map_from_arrays(transform(map_keys(_c2#298268), lambdafunction(lambda key#298408, lambda key#298408, false)), transform(map_values(_c2#298268), lambdafunction(lambda value#298410, lambda value#298410, false))) AS _c2#298412, map_from_arrays(transform(map_keys(_c3#298269), lambdafunction(lambda key#298413, lambda key#298413, false)), transform(map_values(_c3#298269), lambdafunction(lambda value#298415, lambda value#298415, false))) AS _c3#298417, map_from_arrays(transform(map_keys(_c4#298270), lambdafunction(lambda key#298418, lambda key#298418, false)), transform(map_values(_c4#298270), lambdafunction(lambda value#298420, lambda value#298420, false))) AS _c4#298422, map_from_arrays(transform(map_keys(_c5#298271), lambdafunction(lambda key#298423L, lambda key#298423L, false)), transform(map_values(_c5#298271), lambdafunction(lambda value#298425L, lambda value#298425L, false))) AS _c5#298427, map_from_arrays(transform(map_keys(_c6#298272), lambdafunction(lambda key#298428, lambda key#298428, false)), transform(map_values(_c6#298272), lambdafunction(lambda value#298430, lambda value#298430, false))) AS _c6#298432, map_from_arrays(transform(map_keys(_c7#298273), lambdafunction(lambda key#298433, lambda key#298433, false)), transform(map_values(_c7#298273), lambdafunction(lambda value#298435, lambda value#298435, false))) AS _c7#298437, map_from_arrays(transform(map_keys(_c8#298274), lambdafunction(lambda key#298438, lambda key#298438, false)), transform(map_values(_c8#298274), lambdafunction(lambda value#298440, lambda value#298440, false))) AS _c8#298442, map_from_arrays(transform(map_keys(_c9#298275), lambdafunction(lambda key#298443, lambda key#298443, false)), transform(map_values(_c9#298275), lambdafunction(lambda value#298445, lambda value#298445, false))) AS _c9#298447, map_from_arrays(transform(map_keys(_c10#298276), lambdafunction(lambda key#298448, lambda key#298448, false)), transform(map_values(_c10#298276), lambdafunction(lambda value#298450, lambda value#298450, false))) AS _c10#298452, map_from_arrays(transform(map_keys(_c11#298277), lambdafunction(lambda key#298453, lambda key#298453, false)), transform(map_values(_c11#298277), lambdafunction(lambda value#298455, lambda value#298455, false))) AS _c11#298457]

[2024-06-01T00:46:12.165Z] +- Scan ExistingRDD[_c0#298266,_c1#298267,_c2#298268,_c3#298269,_c4#298270,_c5#298271,_c6#298272,_c7#298273,_c8#298274,_c9#298275,_c10#298276,_c11#298277]

[2024-06-01T00:46:12.165Z] FAILED ../../src/main/python/hive_parquet_write_test.py::test_write_compressed_parquet_into_hive_table[UNCOMPRESSED][DATAGEN_SEED=1717194715, TZ=UTC, INJECT_OOM, IGNORE_ORDER({'local': True})] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.165Z] Project [_c0#298626, _c1#298627, _c2#298628, _c3#298629L, _c4#298630, _c5#298631, _c6#298632, _c7#298633, _c8#298634, _c9#298635, _c10#298636, _c11#298637, _c12#298638, if (isnull(_c13#298639)) null else named_struct(c0, _c13#298639.c0, c1, _c13#298639.c1, c2, _c13#298639.c2, c3, _c13#298639.c3, c4, _c13#298639.c4, c5, _c13#298639.c5, c6, _c13#298639.c6, c7, _c13#298639.c7, c8, _c13#298639.c8, c9, _c13#298639.c9, c10, _c13#298639.c10, c11, _c13#298639.c11, ... 2 more fields) AS _c13#299104, if (isnull(_c14#298640)) null else named_struct(child0, _c14#298640.child0, child1, if (isnull(_c14#298640.child1)) null else named_struct(c0, _c14#298640.child1.c0, c1, _c14#298640.child1.c1, c2, _c14#298640.child1.c2, c3, _c14#298640.child1.c3, c4, _c14#298640.child1.c4, c5, _c14#298640.child1.c5, c6, _c14#298640.child1.c6, c7, _c14#298640.child1.c7, c8, _c14#298640.child1.c8, c9, _c14#298640.child1.c9, c10, _c14#298640.child1.c10, c11, _c14#298640.child1.c11, ... 2 more fields)) AS _c14#299136, if (isnull(_c15#298641)) null else named_struct(child0, transform(_c15#298641.child0, lambdafunction(lambda element#299141, lambda element#299141, false)), child1, _c15#298641.child1) AS _c15#299144, transform(_c16#298642, lambdafunction(lambda element#299145, lambda element#299145, false)) AS _c16#299147, transform(_c17#298643, lambdafunction(lambda element#299148, lambda element#299148, false)) AS _c17#299150, transform(_c18#298644, lambdafunction(lambda element#299151, lambda element#299151, false)) AS _c18#299153, transform(_c19#298645, lambdafunction(lambda element#299154L, lambda element#299154L, false)) AS _c19#299156, transform(_c20#298646, lambdafunction(lambda element#299157, lambda element#299157, false)) AS _c20#299159, transform(_c21#298647, lambdafunction(lambda element#299160, lambda element#299160, false)) AS _c21#299162, transform(_c22#298648, lambdafunction(lambda element#299163, lambda element#299163, false)) AS _c22#299165, transform(_c23#298649, lambdafunction(lambda element#299166, lambda element#299166, false)) AS _c23#299168, ... 20 more fields]

[2024-06-01T00:46:12.166Z] +- Scan ExistingRDD[_c0#298626,_c1#298627,_c2#298628,_c3#298629L,_c4#298630,_c5#298631,_c6#298632,_c7#298633,_c8#298634,_c9#298635,_c10#298636,_c11#298637,_c12#298638,_c13#298639,_c14#298640,_c15#298641,_c16#298642,_c17#298643,_c18#298644,_c19#298645,_c20#298646,_c21#298647,_c22#298648,_c23#298649,... 20 more fields]

[2024-06-01T00:46:12.166Z] FAILED ../../src/main/python/hive_parquet_write_test.py::test_write_compressed_parquet_into_hive_table[SNAPPY][DATAGEN_SEED=1717194715, TZ=UTC, INJECT_OOM, IGNORE_ORDER({'local': True})] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.166Z] Project [_c0#299282, _c1#299283, _c2#299284, _c3#299285L, _c4#299286, _c5#299287, _c6#299288, _c7#299289, _c8#299290, _c9#299291, _c10#299292, _c11#299293, _c12#299294, if (isnull(_c13#299295)) null else named_struct(c0, _c13#299295.c0, c1, _c13#299295.c1, c2, _c13#299295.c2, c3, _c13#299295.c3, c4, _c13#299295.c4, c5, _c13#299295.c5, c6, _c13#299295.c6, c7, _c13#299295.c7, c8, _c13#299295.c8, c9, _c13#299295.c9, c10, _c13#299295.c10, c11, _c13#299295.c11, ... 2 more fields) AS _c13#299760, if (isnull(_c14#299296)) null else named_struct(child0, _c14#299296.child0, child1, if (isnull(_c14#299296.child1)) null else named_struct(c0, _c14#299296.child1.c0, c1, _c14#299296.child1.c1, c2, _c14#299296.child1.c2, c3, _c14#299296.child1.c3, c4, _c14#299296.child1.c4, c5, _c14#299296.child1.c5, c6, _c14#299296.child1.c6, c7, _c14#299296.child1.c7, c8, _c14#299296.child1.c8, c9, _c14#299296.child1.c9, c10, _c14#299296.child1.c10, c11, _c14#299296.child1.c11, ... 2 more fields)) AS _c14#299792, if (isnull(_c15#299297)) null else named_struct(child0, transform(_c15#299297.child0, lambdafunction(lambda element#299797, lambda element#299797, false)), child1, _c15#299297.child1) AS _c15#299800, transform(_c16#299298, lambdafunction(lambda element#299801, lambda element#299801, false)) AS _c16#299803, transform(_c17#299299, lambdafunction(lambda element#299804, lambda element#299804, false)) AS _c17#299806, transform(_c18#299300, lambdafunction(lambda element#299807, lambda element#299807, false)) AS _c18#299809, transform(_c19#299301, lambdafunction(lambda element#299810L, lambda element#299810L, false)) AS _c19#299812, transform(_c20#299302, lambdafunction(lambda element#299813, lambda element#299813, false)) AS _c20#299815, transform(_c21#299303, lambdafunction(lambda element#299816, lambda element#299816, false)) AS _c21#299818, transform(_c22#299304, lambdafunction(lambda element#299819, lambda element#299819, false)) AS _c22#299821, transform(_c23#299305, lambdafunction(lambda element#299822, lambda element#299822, false)) AS _c23#299824, ... 20 more fields]

[2024-06-01T00:46:12.166Z] +- Scan ExistingRDD[_c0#299282,_c1#299283,_c2#299284,_c3#299285L,_c4#299286,_c5#299287,_c6#299288,_c7#299289,_c8#299290,_c9#299291,_c10#299292,_c11#299293,_c12#299294,_c13#299295,_c14#299296,_c15#299297,_c16#299298,_c17#299299,_c18#299300,_c19#299301,_c20#299302,_c21#299303,_c22#299304,_c23#299305,... 20 more fields]

[2024-06-01T00:46:12.166Z] FAILED ../../src/main/python/hive_parquet_write_test.py::test_write_compressed_parquet_into_hive_table[ZSTD][DATAGEN_SEED=1717194715, TZ=UTC, IGNORE_ORDER({'local': True})] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.166Z] Project [_c0#299938, _c1#299939, _c2#299940, _c3#299941L, _c4#299942, _c5#299943, _c6#299944, _c7#299945, _c8#299946, _c9#299947, _c10#299948, _c11#299949, _c12#299950, if (isnull(_c13#299951)) null else named_struct(c0, _c13#299951.c0, c1, _c13#299951.c1, c2, _c13#299951.c2, c3, _c13#299951.c3, c4, _c13#299951.c4, c5, _c13#299951.c5, c6, _c13#299951.c6, c7, _c13#299951.c7, c8, _c13#299951.c8, c9, _c13#299951.c9, c10, _c13#299951.c10, c11, _c13#299951.c11, ... 2 more fields) AS _c13#300416, if (isnull(_c14#299952)) null else named_struct(child0, _c14#299952.child0, child1, if (isnull(_c14#299952.child1)) null else named_struct(c0, _c14#299952.child1.c0, c1, _c14#299952.child1.c1, c2, _c14#299952.child1.c2, c3, _c14#299952.child1.c3, c4, _c14#299952.child1.c4, c5, _c14#299952.child1.c5, c6, _c14#299952.child1.c6, c7, _c14#299952.child1.c7, c8, _c14#299952.child1.c8, c9, _c14#299952.child1.c9, c10, _c14#299952.child1.c10, c11, _c14#299952.child1.c11, ... 2 more fields)) AS _c14#300448, if (isnull(_c15#299953)) null else named_struct(child0, transform(_c15#299953.child0, lambdafunction(lambda element#300453, lambda element#300453, false)), child1, _c15#299953.child1) AS _c15#300456, transform(_c16#299954, lambdafunction(lambda element#300457, lambda element#300457, false)) AS _c16#300459, transform(_c17#299955, lambdafunction(lambda element#300460, lambda element#300460, false)) AS _c17#300462, transform(_c18#299956, lambdafunction(lambda element#300463, lambda element#300463, false)) AS _c18#300465, transform(_c19#299957, lambdafunction(lambda element#300466L, lambda element#300466L, false)) AS _c19#300468, transform(_c20#299958, lambdafunction(lambda element#300469, lambda element#300469, false)) AS _c20#300471, transform(_c21#299959, lambdafunction(lambda element#300472, lambda element#300472, false)) AS _c21#300474, transform(_c22#299960, lambdafunction(lambda element#300475, lambda element#300475, false)) AS _c22#300477, transform(_c23#299961, lambdafunction(lambda element#300478, lambda element#300478, false)) AS _c23#300480, ... 20 more fields]

[2024-06-01T00:46:12.166Z] +- Scan ExistingRDD[_c0#299938,_c1#299939,_c2#299940,_c3#299941L,_c4#299942,_c5#299943,_c6#299944,_c7#299945,_c8#299946,_c9#299947,_c10#299948,_c11#299949,_c12#299950,_c13#299951,_c14#299952,_c15#299953,_c16#299954,_c17#299955,_c18#299956,_c19#299957,_c20#299958,_c21#299959,_c22#299960,_c23#299961,... 20 more fields]

[2024-06-01T00:46:12.166Z] = 4 failed, 28068 passed, 955 skipped, 669 xfailed, 920 xpassed, 15696 warnings in 8055.73s (2:14:15) =
```

<details>
<summary> Details of test failures </summary>
<pre>
[2024-06-01T00:46:12.158Z] =================================== FAILURES ===================================

[2024-06-01T00:46:12.158Z] _ test_write_parquet_into_hive_table[[Map(String(not_null),String), Map(Boolean(not_null),Boolean), Map(Byte(not_null),Byte), Map(Short(not_null),Short), Map(Integer(not_null),Integer), Map(Long(not_null),Long), Map(Float(not_null),Float), Map(Double(not_null),Double), Map(Timestamp(not_null),Timestamp), Map(Date(not_null),Date), Map(Decimal(not_null)(19,1),Decimal(19,1)), Map(Decimal(not_null)(36,5),Decimal(36,5))]-CTTW] _

[2024-06-01T00:46:12.158Z] [gw1] linux -- Python 3.9.19 /opt/conda/bin/python

[2024-06-01T00:46:12.158Z] 

[2024-06-01T00:46:12.158Z] spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f7fe2f3d280>

[2024-06-01T00:46:12.158Z] is_ctas = False

[2024-06-01T00:46:12.158Z] gens = [Map(String(not_null),String), Map(Boolean(not_null),Boolean), Map(Byte(not_null),Byte), Map(Short(not_null),Short), Map(Integer(not_null),Integer), Map(Long(not_null),Long), ...]

[2024-06-01T00:46:12.158Z] 

[2024-06-01T00:46:12.158Z]     @allow_non_gpu(*(non_utc_allow + fallback_nodes))

[2024-06-01T00:46:12.158Z]     @ignore_order(local=True)

[2024-06-01T00:46:12.158Z]     @pytest.mark.parametrize("is_ctas", [True, False], ids=['CTAS', 'CTTW'])

[2024-06-01T00:46:12.158Z]     @pytest.mark.parametrize("gens", _hive_write_gens, ids=idfn)

[2024-06-01T00:46:12.158Z]     def test_write_parquet_into_hive_table(spark_tmp_table_factory, is_ctas, gens):

[2024-06-01T00:46:12.158Z]     

[2024-06-01T00:46:12.158Z]         def gen_table(spark):

[2024-06-01T00:46:12.158Z]             gen_list = [('_c' + str(i), gen) for i, gen in enumerate(gens)]

[2024-06-01T00:46:12.158Z]             types_sql_str = ','.join('{} {}'.format(

[2024-06-01T00:46:12.158Z]                 name, gen.data_type.simpleString()) for name, gen in gen_list)

[2024-06-01T00:46:12.158Z]             data_table = spark_tmp_table_factory.get()

[2024-06-01T00:46:12.158Z]             gen_df(spark, gen_list).createOrReplaceTempView(data_table)

[2024-06-01T00:46:12.158Z]             return data_table, types_sql_str

[2024-06-01T00:46:12.158Z]     

[2024-06-01T00:46:12.158Z]         (input_table, input_schema) = with_cpu_session(gen_table)

[2024-06-01T00:46:12.158Z]     

[2024-06-01T00:46:12.158Z]         def write_to_hive_sql(spark, output_table):

[2024-06-01T00:46:12.158Z]             if is_ctas:

[2024-06-01T00:46:12.158Z]                 # Create Table As Select

[2024-06-01T00:46:12.158Z]                 return [

[2024-06-01T00:46:12.158Z]                     "CREATE TABLE {} STORED AS PARQUET AS SELECT * FROM {}".format(

[2024-06-01T00:46:12.158Z]                         output_table, input_table)

[2024-06-01T00:46:12.158Z]                 ]

[2024-06-01T00:46:12.158Z]             else:

[2024-06-01T00:46:12.158Z]                 # Create Table Then Write

[2024-06-01T00:46:12.158Z]                 return [

[2024-06-01T00:46:12.158Z]                     "CREATE TABLE {} ({}) STORED AS PARQUET".format(output_table, input_schema),

[2024-06-01T00:46:12.158Z]                     "INSERT OVERWRITE TABLE {} SELECT * FROM {}".format(output_table, input_table)

[2024-06-01T00:46:12.158Z]                 ]

[2024-06-01T00:46:12.158Z]     

[2024-06-01T00:46:12.158Z] >       assert_gpu_and_cpu_sql_writes_are_equal_collect(

[2024-06-01T00:46:12.158Z]             spark_tmp_table_factory,

[2024-06-01T00:46:12.158Z]             write_to_hive_sql,

[2024-06-01T00:46:12.158Z]             _write_to_hive_conf)

[2024-06-01T00:46:12.158Z] 

[2024-06-01T00:46:12.158Z] ../../src/main/python/hive_parquet_write_test.py:91: 

[2024-06-01T00:46:12.158Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.158Z] ../../src/main/python/asserts.py:319: in assert_gpu_and_cpu_sql_writes_are_equal_collect

[2024-06-01T00:46:12.158Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.158Z] ../../src/main/python/spark_session.py:164: in with_gpu_session

[2024-06-01T00:46:12.158Z]     return with_spark_session(func, conf=copy)

[2024-06-01T00:46:12.159Z] /opt/conda/lib/python3.9/contextlib.py:79: in inner

[2024-06-01T00:46:12.159Z]     return func(*args, **kwds)

[2024-06-01T00:46:12.159Z] ../../src/main/python/spark_session.py:131: in with_spark_session

[2024-06-01T00:46:12.159Z]     ret = func(_spark)

[2024-06-01T00:46:12.159Z] ../../src/main/python/asserts.py:319: in <lambda>

[2024-06-01T00:46:12.159Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.159Z] ../../src/main/python/asserts.py:312: in do_write

[2024-06-01T00:46:12.159Z]     spark.sql(sql_text)

[2024-06-01T00:46:12.159Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/sql/session.py:1631: in sql

[2024-06-01T00:46:12.159Z]     return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)

[2024-06-01T00:46:12.159Z] /home/jenkins/agent/workspace/jenkins-rapids_integration-dev-github-1001-3.5.1/jars/spark-3.5.1-bin-hadoop3/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py:1322: in __call__

[2024-06-01T00:46:12.159Z]     return_value = get_return_value(

[2024-06-01T00:46:12.159Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.159Z] 

[2024-06-01T00:46:12.159Z] a = ('xro745325', <py4j.clientserver.JavaClient object at 0x7f7fffb77ca0>, 'o80', 'sql')

[2024-06-01T00:46:12.159Z] kw = {}, converted = IllegalArgumentException()

[2024-06-01T00:46:12.159Z] 

[2024-06-01T00:46:12.159Z]     def deco(*a: Any, **kw: Any) -> Any:

[2024-06-01T00:46:12.159Z]         try:

[2024-06-01T00:46:12.159Z]             return f(*a, **kw)

[2024-06-01T00:46:12.159Z]         except Py4JJavaError as e:

[2024-06-01T00:46:12.159Z]             converted = convert_exception(e.java_exception)

[2024-06-01T00:46:12.159Z]             if not isinstance(converted, UnknownException):

[2024-06-01T00:46:12.159Z]                 # Hide where the exception came from that shows a non-Pythonic

[2024-06-01T00:46:12.159Z]                 # JVM exception message.

[2024-06-01T00:46:12.159Z] >               raise converted from None

[2024-06-01T00:46:12.159Z] E               pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.159Z] E               Project [map_from_arrays(transform(map_keys(_c0#298266), lambdafunction(lambda key#298398, lambda key#298398, false)), transform(map_values(_c0#298266), lambdafunction(lambda value#298400, lambda value#298400, false))) AS _c0#298402, map_from_arrays(transform(map_keys(_c1#298267), lambdafunction(lambda key#298403, lambda key#298403, false)), transform(map_values(_c1#298267), lambdafunction(lambda value#298405, lambda value#298405, false))) AS _c1#298407, map_from_arrays(transform(map_keys(_c2#298268), lambdafunction(lambda key#298408, lambda key#298408, false)), transform(map_values(_c2#298268), lambdafunction(lambda value#298410, lambda value#298410, false))) AS _c2#298412, map_from_arrays(transform(map_keys(_c3#298269), lambdafunction(lambda key#298413, lambda key#298413, false)), transform(map_values(_c3#298269), lambdafunction(lambda value#298415, lambda value#298415, false))) AS _c3#298417, map_from_arrays(transform(map_keys(_c4#298270), lambdafunction(lambda key#298418, lambda key#298418, false)), transform(map_values(_c4#298270), lambdafunction(lambda value#298420, lambda value#298420, false))) AS _c4#298422, map_from_arrays(transform(map_keys(_c5#298271), lambdafunction(lambda key#298423L, lambda key#298423L, false)), transform(map_values(_c5#298271), lambdafunction(lambda value#298425L, lambda value#298425L, false))) AS _c5#298427, map_from_arrays(transform(map_keys(_c6#298272), lambdafunction(lambda key#298428, lambda key#298428, false)), transform(map_values(_c6#298272), lambdafunction(lambda value#298430, lambda value#298430, false))) AS _c6#298432, map_from_arrays(transform(map_keys(_c7#298273), lambdafunction(lambda key#298433, lambda key#298433, false)), transform(map_values(_c7#298273), lambdafunction(lambda value#298435, lambda value#298435, false))) AS _c7#298437, map_from_arrays(transform(map_keys(_c8#298274), lambdafunction(lambda key#298438, lambda key#298438, false)), transform(map_values(_c8#298274), lambdafunction(lambda value#298440, lambda value#298440, false))) AS _c8#298442, map_from_arrays(transform(map_keys(_c9#298275), lambdafunction(lambda key#298443, lambda key#298443, false)), transform(map_values(_c9#298275), lambdafunction(lambda value#298445, lambda value#298445, false))) AS _c9#298447, map_from_arrays(transform(map_keys(_c10#298276), lambdafunction(lambda key#298448, lambda key#298448, false)), transform(map_values(_c10#298276), lambdafunction(lambda value#298450, lambda value#298450, false))) AS _c10#298452, map_from_arrays(transform(map_keys(_c11#298277), lambdafunction(lambda key#298453, lambda key#298453, false)), transform(map_values(_c11#298277), lambdafunction(lambda value#298455, lambda value#298455, false))) AS _c11#298457]

[2024-06-01T00:46:12.159Z] E               +- Scan ExistingRDD[_c0#298266,_c1#298267,_c2#298268,_c3#298269,_c4#298270,_c5#298271,_c6#298272,_c7#298273,_c8#298274,_c9#298275,_c10#298276,_c11#298277]

[2024-06-01T00:46:12.159Z] 

[2024-06-01T00:46:12.159Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/errors/exceptions/captured.py:185: IllegalArgumentException

[2024-06-01T00:46:12.159Z] ----------------------------- Captured stdout call -----------------------------

[2024-06-01T00:46:12.159Z] ### CPU RUN ###

[2024-06-01T00:46:12.159Z] ### GPU RUN ###

[2024-06-01T00:46:12.159Z] _________ test_write_compressed_parquet_into_hive_table[UNCOMPRESSED] __________

[2024-06-01T00:46:12.159Z] [gw1] linux -- Python 3.9.19 /opt/conda/bin/python

[2024-06-01T00:46:12.159Z] 

[2024-06-01T00:46:12.159Z] spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f7fe7530e20>

[2024-06-01T00:46:12.159Z] comp_type = 'UNCOMPRESSED'

[2024-06-01T00:46:12.159Z] 

[2024-06-01T00:46:12.159Z]     @allow_non_gpu(*(non_utc_allow + fallback_nodes))

[2024-06-01T00:46:12.159Z]     @ignore_order(local=True)

[2024-06-01T00:46:12.159Z]     @pytest.mark.parametrize("comp_type", ['UNCOMPRESSED', 'SNAPPY', zstd_param])

[2024-06-01T00:46:12.159Z]     def test_write_compressed_parquet_into_hive_table(spark_tmp_table_factory, comp_type):

[2024-06-01T00:46:12.159Z]         # Generate hive table in Parquet format

[2024-06-01T00:46:12.159Z]         def gen_table(spark):

[2024-06-01T00:46:12.159Z]             gens = _hive_basic_gens + _hive_struct_gens + _hive_array_gens + _hive_map_gens

[2024-06-01T00:46:12.159Z]             gen_list = [('_c' + str(i), gen) for i, gen in enumerate(gens)]

[2024-06-01T00:46:12.159Z]             types_sql_str = ','.join('{} {}'.format(

[2024-06-01T00:46:12.160Z]                 name, gen.data_type.simpleString()) for name, gen in gen_list)

[2024-06-01T00:46:12.160Z]             data_table = spark_tmp_table_factory.get()

[2024-06-01T00:46:12.160Z]             gen_df(spark, gen_list).createOrReplaceTempView(data_table)

[2024-06-01T00:46:12.160Z]             return data_table, types_sql_str

[2024-06-01T00:46:12.160Z]     

[2024-06-01T00:46:12.160Z]         input_table, schema_str = with_cpu_session(gen_table)

[2024-06-01T00:46:12.160Z]     

[2024-06-01T00:46:12.160Z]         def write_to_hive_sql(spark, output_table):

[2024-06-01T00:46:12.160Z]             return [

[2024-06-01T00:46:12.160Z]                 # Create table with compression type

[2024-06-01T00:46:12.160Z]                 "CREATE TABLE {} ({}) STORED AS PARQUET "

[2024-06-01T00:46:12.160Z]                 "TBLPROPERTIES ('parquet.compression'='{}')".format(

[2024-06-01T00:46:12.160Z]                     output_table, schema_str, comp_type),

[2024-06-01T00:46:12.160Z]                 # Insert into table

[2024-06-01T00:46:12.160Z]                 "INSERT OVERWRITE TABLE {} SELECT * FROM {}".format(output_table, input_table)

[2024-06-01T00:46:12.160Z]             ]

[2024-06-01T00:46:12.160Z]     

[2024-06-01T00:46:12.160Z] >       assert_gpu_and_cpu_sql_writes_are_equal_collect(

[2024-06-01T00:46:12.160Z]             spark_tmp_table_factory,

[2024-06-01T00:46:12.160Z]             write_to_hive_sql,

[2024-06-01T00:46:12.160Z]             _write_to_hive_conf)

[2024-06-01T00:46:12.160Z] 

[2024-06-01T00:46:12.160Z] ../../src/main/python/hive_parquet_write_test.py:173: 

[2024-06-01T00:46:12.160Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.160Z] ../../src/main/python/asserts.py:319: in assert_gpu_and_cpu_sql_writes_are_equal_collect

[2024-06-01T00:46:12.160Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.160Z] ../../src/main/python/spark_session.py:164: in with_gpu_session

[2024-06-01T00:46:12.160Z]     return with_spark_session(func, conf=copy)

[2024-06-01T00:46:12.160Z] /opt/conda/lib/python3.9/contextlib.py:79: in inner

[2024-06-01T00:46:12.160Z]     return func(*args, **kwds)

[2024-06-01T00:46:12.160Z] ../../src/main/python/spark_session.py:131: in with_spark_session

[2024-06-01T00:46:12.160Z]     ret = func(_spark)

[2024-06-01T00:46:12.160Z] ../../src/main/python/asserts.py:319: in <lambda>

[2024-06-01T00:46:12.160Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.160Z] ../../src/main/python/asserts.py:312: in do_write

[2024-06-01T00:46:12.160Z]     spark.sql(sql_text)

[2024-06-01T00:46:12.160Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/sql/session.py:1631: in sql

[2024-06-01T00:46:12.160Z]     return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)

[2024-06-01T00:46:12.160Z] /home/jenkins/agent/workspace/jenkins-rapids_integration-dev-github-1001-3.5.1/jars/spark-3.5.1-bin-hadoop3/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py:1322: in __call__

[2024-06-01T00:46:12.160Z]     return_value = get_return_value(

[2024-06-01T00:46:12.160Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.160Z] 

[2024-06-01T00:46:12.160Z] a = ('xro746483', <py4j.clientserver.JavaClient object at 0x7f7fffb77ca0>, 'o80', 'sql')

[2024-06-01T00:46:12.160Z] kw = {}, converted = IllegalArgumentException()

[2024-06-01T00:46:12.160Z] 

[2024-06-01T00:46:12.160Z]     def deco(*a: Any, **kw: Any) -> Any:

[2024-06-01T00:46:12.160Z]         try:

[2024-06-01T00:46:12.160Z]             return f(*a, **kw)

[2024-06-01T00:46:12.160Z]         except Py4JJavaError as e:

[2024-06-01T00:46:12.160Z]             converted = convert_exception(e.java_exception)

[2024-06-01T00:46:12.160Z]             if not isinstance(converted, UnknownException):

[2024-06-01T00:46:12.160Z]                 # Hide where the exception came from that shows a non-Pythonic

[2024-06-01T00:46:12.160Z]                 # JVM exception message.

[2024-06-01T00:46:12.160Z] >               raise converted from None

[2024-06-01T00:46:12.160Z] E               pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.160Z] E               Project [_c0#298626, _c1#298627, _c2#298628, _c3#298629L, _c4#298630, _c5#298631, _c6#298632, _c7#298633, _c8#298634, _c9#298635, _c10#298636, _c11#298637, _c12#298638, if (isnull(_c13#298639)) null else named_struct(c0, _c13#298639.c0, c1, _c13#298639.c1, c2, _c13#298639.c2, c3, _c13#298639.c3, c4, _c13#298639.c4, c5, _c13#298639.c5, c6, _c13#298639.c6, c7, _c13#298639.c7, c8, _c13#298639.c8, c9, _c13#298639.c9, c10, _c13#298639.c10, c11, _c13#298639.c11, ... 2 more fields) AS _c13#299104, if (isnull(_c14#298640)) null else named_struct(child0, _c14#298640.child0, child1, if (isnull(_c14#298640.child1)) null else named_struct(c0, _c14#298640.child1.c0, c1, _c14#298640.child1.c1, c2, _c14#298640.child1.c2, c3, _c14#298640.child1.c3, c4, _c14#298640.child1.c4, c5, _c14#298640.child1.c5, c6, _c14#298640.child1.c6, c7, _c14#298640.child1.c7, c8, _c14#298640.child1.c8, c9, _c14#298640.child1.c9, c10, _c14#298640.child1.c10, c11, _c14#298640.child1.c11, ... 2 more fields)) AS _c14#299136, if (isnull(_c15#298641)) null else named_struct(child0, transform(_c15#298641.child0, lambdafunction(lambda element#299141, lambda element#299141, false)), child1, _c15#298641.child1) AS _c15#299144, transform(_c16#298642, lambdafunction(lambda element#299145, lambda element#299145, false)) AS _c16#299147, transform(_c17#298643, lambdafunction(lambda element#299148, lambda element#299148, false)) AS _c17#299150, transform(_c18#298644, lambdafunction(lambda element#299151, lambda element#299151, false)) AS _c18#299153, transform(_c19#298645, lambdafunction(lambda element#299154L, lambda element#299154L, false)) AS _c19#299156, transform(_c20#298646, lambdafunction(lambda element#299157, lambda element#299157, false)) AS _c20#299159, transform(_c21#298647, lambdafunction(lambda element#299160, lambda element#299160, false)) AS _c21#299162, transform(_c22#298648, lambdafunction(lambda element#299163, lambda element#299163, false)) AS _c22#299165, transform(_c23#298649, lambdafunction(lambda element#299166, lambda element#299166, false)) AS _c23#299168, ... 20 more fields]

[2024-06-01T00:46:12.161Z] E               +- Scan ExistingRDD[_c0#298626,_c1#298627,_c2#298628,_c3#298629L,_c4#298630,_c5#298631,_c6#298632,_c7#298633,_c8#298634,_c9#298635,_c10#298636,_c11#298637,_c12#298638,_c13#298639,_c14#298640,_c15#298641,_c16#298642,_c17#298643,_c18#298644,_c19#298645,_c20#298646,_c21#298647,_c22#298648,_c23#298649,... 20 more fields]

[2024-06-01T00:46:12.161Z] 

[2024-06-01T00:46:12.161Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/errors/exceptions/captured.py:185: IllegalArgumentException

[2024-06-01T00:46:12.161Z] ----------------------------- Captured stdout call -----------------------------

[2024-06-01T00:46:12.161Z] ### CPU RUN ###

[2024-06-01T00:46:12.161Z] ### GPU RUN ###

[2024-06-01T00:46:12.161Z] ____________ test_write_compressed_parquet_into_hive_table[SNAPPY] _____________

[2024-06-01T00:46:12.161Z] [gw1] linux -- Python 3.9.19 /opt/conda/bin/python

[2024-06-01T00:46:12.161Z] 

[2024-06-01T00:46:12.161Z] spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f7fd835df10>

[2024-06-01T00:46:12.161Z] comp_type = 'SNAPPY'

[2024-06-01T00:46:12.161Z] 

[2024-06-01T00:46:12.161Z]     @allow_non_gpu(*(non_utc_allow + fallback_nodes))

[2024-06-01T00:46:12.161Z]     @ignore_order(local=True)

[2024-06-01T00:46:12.161Z]     @pytest.mark.parametrize("comp_type", ['UNCOMPRESSED', 'SNAPPY', zstd_param])

[2024-06-01T00:46:12.161Z]     def test_write_compressed_parquet_into_hive_table(spark_tmp_table_factory, comp_type):

[2024-06-01T00:46:12.161Z]         # Generate hive table in Parquet format

[2024-06-01T00:46:12.161Z]         def gen_table(spark):

[2024-06-01T00:46:12.161Z]             gens = _hive_basic_gens + _hive_struct_gens + _hive_array_gens + _hive_map_gens

[2024-06-01T00:46:12.161Z]             gen_list = [('_c' + str(i), gen) for i, gen in enumerate(gens)]

[2024-06-01T00:46:12.161Z]             types_sql_str = ','.join('{} {}'.format(

[2024-06-01T00:46:12.161Z]                 name, gen.data_type.simpleString()) for name, gen in gen_list)

[2024-06-01T00:46:12.161Z]             data_table = spark_tmp_table_factory.get()

[2024-06-01T00:46:12.161Z]             gen_df(spark, gen_list).createOrReplaceTempView(data_table)

[2024-06-01T00:46:12.161Z]             return data_table, types_sql_str

[2024-06-01T00:46:12.161Z]     

[2024-06-01T00:46:12.161Z]         input_table, schema_str = with_cpu_session(gen_table)

[2024-06-01T00:46:12.161Z]     

[2024-06-01T00:46:12.161Z]         def write_to_hive_sql(spark, output_table):

[2024-06-01T00:46:12.161Z]             return [

[2024-06-01T00:46:12.161Z]                 # Create table with compression type

[2024-06-01T00:46:12.161Z]                 "CREATE TABLE {} ({}) STORED AS PARQUET "

[2024-06-01T00:46:12.161Z]                 "TBLPROPERTIES ('parquet.compression'='{}')".format(

[2024-06-01T00:46:12.161Z]                     output_table, schema_str, comp_type),

[2024-06-01T00:46:12.161Z]                 # Insert into table

[2024-06-01T00:46:12.161Z]                 "INSERT OVERWRITE TABLE {} SELECT * FROM {}".format(output_table, input_table)

[2024-06-01T00:46:12.161Z]             ]

[2024-06-01T00:46:12.161Z]     

[2024-06-01T00:46:12.161Z] >       assert_gpu_and_cpu_sql_writes_are_equal_collect(

[2024-06-01T00:46:12.161Z]             spark_tmp_table_factory,

[2024-06-01T00:46:12.161Z]             write_to_hive_sql,

[2024-06-01T00:46:12.161Z]             _write_to_hive_conf)

[2024-06-01T00:46:12.161Z] 

[2024-06-01T00:46:12.161Z] ../../src/main/python/hive_parquet_write_test.py:173: 

[2024-06-01T00:46:12.161Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.161Z] ../../src/main/python/asserts.py:319: in assert_gpu_and_cpu_sql_writes_are_equal_collect

[2024-06-01T00:46:12.161Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.161Z] ../../src/main/python/spark_session.py:164: in with_gpu_session

[2024-06-01T00:46:12.161Z]     return with_spark_session(func, conf=copy)

[2024-06-01T00:46:12.161Z] /opt/conda/lib/python3.9/contextlib.py:79: in inner

[2024-06-01T00:46:12.161Z]     return func(*args, **kwds)

[2024-06-01T00:46:12.161Z] ../../src/main/python/spark_session.py:131: in with_spark_session

[2024-06-01T00:46:12.161Z]     ret = func(_spark)

[2024-06-01T00:46:12.161Z] ../../src/main/python/asserts.py:319: in <lambda>

[2024-06-01T00:46:12.161Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.161Z] ../../src/main/python/asserts.py:312: in do_write

[2024-06-01T00:46:12.161Z]     spark.sql(sql_text)

[2024-06-01T00:46:12.161Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/sql/session.py:1631: in sql

[2024-06-01T00:46:12.161Z]     return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)

[2024-06-01T00:46:12.161Z] /home/jenkins/agent/workspace/jenkins-rapids_integration-dev-github-1001-3.5.1/jars/spark-3.5.1-bin-hadoop3/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py:1322: in __call__

[2024-06-01T00:46:12.161Z]     return_value = get_return_value(

[2024-06-01T00:46:12.161Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.161Z] 

[2024-06-01T00:46:12.161Z] a = ('xro746760', <py4j.clientserver.JavaClient object at 0x7f7fffb77ca0>, 'o80', 'sql')

[2024-06-01T00:46:12.161Z] kw = {}, converted = IllegalArgumentException()

[2024-06-01T00:46:12.161Z] 

[2024-06-01T00:46:12.161Z]     def deco(*a: Any, **kw: Any) -> Any:

[2024-06-01T00:46:12.161Z]         try:

[2024-06-01T00:46:12.161Z]             return f(*a, **kw)

[2024-06-01T00:46:12.161Z]         except Py4JJavaError as e:

[2024-06-01T00:46:12.161Z]             converted = convert_exception(e.java_exception)

[2024-06-01T00:46:12.161Z]             if not isinstance(converted, UnknownException):

[2024-06-01T00:46:12.161Z]                 # Hide where the exception came from that shows a non-Pythonic

[2024-06-01T00:46:12.161Z]                 # JVM exception message.

[2024-06-01T00:46:12.161Z] >               raise converted from None

[2024-06-01T00:46:12.162Z] E               pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.162Z] E               Project [_c0#299282, _c1#299283, _c2#299284, _c3#299285L, _c4#299286, _c5#299287, _c6#299288, _c7#299289, _c8#299290, _c9#299291, _c10#299292, _c11#299293, _c12#299294, if (isnull(_c13#299295)) null else named_struct(c0, _c13#299295.c0, c1, _c13#299295.c1, c2, _c13#299295.c2, c3, _c13#299295.c3, c4, _c13#299295.c4, c5, _c13#299295.c5, c6, _c13#299295.c6, c7, _c13#299295.c7, c8, _c13#299295.c8, c9, _c13#299295.c9, c10, _c13#299295.c10, c11, _c13#299295.c11, ... 2 more fields) AS _c13#299760, if (isnull(_c14#299296)) null else named_struct(child0, _c14#299296.child0, child1, if (isnull(_c14#299296.child1)) null else named_struct(c0, _c14#299296.child1.c0, c1, _c14#299296.child1.c1, c2, _c14#299296.child1.c2, c3, _c14#299296.child1.c3, c4, _c14#299296.child1.c4, c5, _c14#299296.child1.c5, c6, _c14#299296.child1.c6, c7, _c14#299296.child1.c7, c8, _c14#299296.child1.c8, c9, _c14#299296.child1.c9, c10, _c14#299296.child1.c10, c11, _c14#299296.child1.c11, ... 2 more fields)) AS _c14#299792, if (isnull(_c15#299297)) null else named_struct(child0, transform(_c15#299297.child0, lambdafunction(lambda element#299797, lambda element#299797, false)), child1, _c15#299297.child1) AS _c15#299800, transform(_c16#299298, lambdafunction(lambda element#299801, lambda element#299801, false)) AS _c16#299803, transform(_c17#299299, lambdafunction(lambda element#299804, lambda element#299804, false)) AS _c17#299806, transform(_c18#299300, lambdafunction(lambda element#299807, lambda element#299807, false)) AS _c18#299809, transform(_c19#299301, lambdafunction(lambda element#299810L, lambda element#299810L, false)) AS _c19#299812, transform(_c20#299302, lambdafunction(lambda element#299813, lambda element#299813, false)) AS _c20#299815, transform(_c21#299303, lambdafunction(lambda element#299816, lambda element#299816, false)) AS _c21#299818, transform(_c22#299304, lambdafunction(lambda element#299819, lambda element#299819, false)) AS _c22#299821, transform(_c23#299305, lambdafunction(lambda element#299822, lambda element#299822, false)) AS _c23#299824, ... 20 more fields]

[2024-06-01T00:46:12.162Z] E               +- Scan ExistingRDD[_c0#299282,_c1#299283,_c2#299284,_c3#299285L,_c4#299286,_c5#299287,_c6#299288,_c7#299289,_c8#299290,_c9#299291,_c10#299292,_c11#299293,_c12#299294,_c13#299295,_c14#299296,_c15#299297,_c16#299298,_c17#299299,_c18#299300,_c19#299301,_c20#299302,_c21#299303,_c22#299304,_c23#299305,... 20 more fields]

[2024-06-01T00:46:12.162Z] 

[2024-06-01T00:46:12.162Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/errors/exceptions/captured.py:185: IllegalArgumentException

[2024-06-01T00:46:12.162Z] ----------------------------- Captured stdout call -----------------------------

[2024-06-01T00:46:12.162Z] ### CPU RUN ###

[2024-06-01T00:46:12.162Z] ### GPU RUN ###

[2024-06-01T00:46:12.162Z] _____________ test_write_compressed_parquet_into_hive_table[ZSTD] ______________

[2024-06-01T00:46:12.162Z] [gw1] linux -- Python 3.9.19 /opt/conda/bin/python

[2024-06-01T00:46:12.162Z] 

[2024-06-01T00:46:12.162Z] spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f7fd8355f70>

[2024-06-01T00:46:12.162Z] comp_type = 'ZSTD'

[2024-06-01T00:46:12.162Z] 

[2024-06-01T00:46:12.162Z]     @allow_non_gpu(*(non_utc_allow + fallback_nodes))

[2024-06-01T00:46:12.162Z]     @ignore_order(local=True)

[2024-06-01T00:46:12.162Z]     @pytest.mark.parametrize("comp_type", ['UNCOMPRESSED', 'SNAPPY', zstd_param])

[2024-06-01T00:46:12.162Z]     def test_write_compressed_parquet_into_hive_table(spark_tmp_table_factory, comp_type):

[2024-06-01T00:46:12.162Z]         # Generate hive table in Parquet format

[2024-06-01T00:46:12.162Z]         def gen_table(spark):

[2024-06-01T00:46:12.162Z]             gens = _hive_basic_gens + _hive_struct_gens + _hive_array_gens + _hive_map_gens

[2024-06-01T00:46:12.162Z]             gen_list = [('_c' + str(i), gen) for i, gen in enumerate(gens)]

[2024-06-01T00:46:12.162Z]             types_sql_str = ','.join('{} {}'.format(

[2024-06-01T00:46:12.162Z]                 name, gen.data_type.simpleString()) for name, gen in gen_list)

[2024-06-01T00:46:12.162Z]             data_table = spark_tmp_table_factory.get()

[2024-06-01T00:46:12.162Z]             gen_df(spark, gen_list).createOrReplaceTempView(data_table)

[2024-06-01T00:46:12.162Z]             return data_table, types_sql_str

[2024-06-01T00:46:12.162Z]     

[2024-06-01T00:46:12.162Z]         input_table, schema_str = with_cpu_session(gen_table)

[2024-06-01T00:46:12.162Z]     

[2024-06-01T00:46:12.162Z]         def write_to_hive_sql(spark, output_table):

[2024-06-01T00:46:12.162Z]             return [

[2024-06-01T00:46:12.162Z]                 # Create table with compression type

[2024-06-01T00:46:12.162Z]                 "CREATE TABLE {} ({}) STORED AS PARQUET "

[2024-06-01T00:46:12.162Z]                 "TBLPROPERTIES ('parquet.compression'='{}')".format(

[2024-06-01T00:46:12.162Z]                     output_table, schema_str, comp_type),

[2024-06-01T00:46:12.162Z]                 # Insert into table

[2024-06-01T00:46:12.162Z]                 "INSERT OVERWRITE TABLE {} SELECT * FROM {}".format(output_table, input_table)

[2024-06-01T00:46:12.162Z]             ]

[2024-06-01T00:46:12.162Z]     

[2024-06-01T00:46:12.162Z] >       assert_gpu_and_cpu_sql_writes_are_equal_collect(

[2024-06-01T00:46:12.162Z]             spark_tmp_table_factory,

[2024-06-01T00:46:12.162Z]             write_to_hive_sql,

[2024-06-01T00:46:12.162Z]             _write_to_hive_conf)

[2024-06-01T00:46:12.162Z] 

[2024-06-01T00:46:12.162Z] ../../src/main/python/hive_parquet_write_test.py:173: 

[2024-06-01T00:46:12.162Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.162Z] ../../src/main/python/asserts.py:319: in assert_gpu_and_cpu_sql_writes_are_equal_collect

[2024-06-01T00:46:12.162Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.162Z] ../../src/main/python/spark_session.py:164: in with_gpu_session

[2024-06-01T00:46:12.162Z]     return with_spark_session(func, conf=copy)

[2024-06-01T00:46:12.162Z] /opt/conda/lib/python3.9/contextlib.py:79: in inner

[2024-06-01T00:46:12.162Z]     return func(*args, **kwds)

[2024-06-01T00:46:12.163Z] ../../src/main/python/spark_session.py:131: in with_spark_session

[2024-06-01T00:46:12.163Z]     ret = func(_spark)

[2024-06-01T00:46:12.163Z] ../../src/main/python/asserts.py:319: in <lambda>

[2024-06-01T00:46:12.163Z]     with_gpu_session(lambda spark : do_write(spark, gpu_table), conf=conf)

[2024-06-01T00:46:12.163Z] ../../src/main/python/asserts.py:312: in do_write

[2024-06-01T00:46:12.163Z]     spark.sql(sql_text)

[2024-06-01T00:46:12.163Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/sql/session.py:1631: in sql

[2024-06-01T00:46:12.163Z]     return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)

[2024-06-01T00:46:12.163Z] /home/jenkins/agent/workspace/jenkins-rapids_integration-dev-github-1001-3.5.1/jars/spark-3.5.1-bin-hadoop3/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py:1322: in __call__

[2024-06-01T00:46:12.163Z]     return_value = get_return_value(

[2024-06-01T00:46:12.163Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-06-01T00:46:12.163Z] 

[2024-06-01T00:46:12.163Z] a = ('xro747035', <py4j.clientserver.JavaClient object at 0x7f7fffb77ca0>, 'o80', 'sql')

[2024-06-01T00:46:12.163Z] kw = {}, converted = IllegalArgumentException()

[2024-06-01T00:46:12.163Z] 

[2024-06-01T00:46:12.163Z]     def deco(*a: Any, **kw: Any) -> Any:

[2024-06-01T00:46:12.163Z]         try:

[2024-06-01T00:46:12.163Z]             return f(*a, **kw)

[2024-06-01T00:46:12.163Z]         except Py4JJavaError as e:

[2024-06-01T00:46:12.163Z]             converted = convert_exception(e.java_exception)

[2024-06-01T00:46:12.163Z]             if not isinstance(converted, UnknownException):

[2024-06-01T00:46:12.163Z]                 # Hide where the exception came from that shows a non-Pythonic

[2024-06-01T00:46:12.163Z]                 # JVM exception message.

[2024-06-01T00:46:12.163Z] >               raise converted from None

[2024-06-01T00:46:12.163Z] E               pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2024-06-01T00:46:12.163Z] E               Project [_c0#299938, _c1#299939, _c2#299940, _c3#299941L, _c4#299942, _c5#299943, _c6#299944, _c7#299945, _c8#299946, _c9#299947, _c10#299948, _c11#299949, _c12#299950, if (isnull(_c13#299951)) null else named_struct(c0, _c13#299951.c0, c1, _c13#299951.c1, c2, _c13#299951.c2, c3, _c13#299951.c3, c4, _c13#299951.c4, c5, _c13#299951.c5, c6, _c13#299951.c6, c7, _c13#299951.c7, c8, _c13#299951.c8, c9, _c13#299951.c9, c10, _c13#299951.c10, c11, _c13#299951.c11, ... 2 more fields) AS _c13#300416, if (isnull(_c14#299952)) null else named_struct(child0, _c14#299952.child0, child1, if (isnull(_c14#299952.child1)) null else named_struct(c0, _c14#299952.child1.c0, c1, _c14#299952.child1.c1, c2, _c14#299952.child1.c2, c3, _c14#299952.child1.c3, c4, _c14#299952.child1.c4, c5, _c14#299952.child1.c5, c6, _c14#299952.child1.c6, c7, _c14#299952.child1.c7, c8, _c14#299952.child1.c8, c9, _c14#299952.child1.c9, c10, _c14#299952.child1.c10, c11, _c14#299952.child1.c11, ... 2 more fields)) AS _c14#300448, if (isnull(_c15#299953)) null else named_struct(child0, transform(_c15#299953.child0, lambdafunction(lambda element#300453, lambda element#300453, false)), child1, _c15#299953.child1) AS _c15#300456, transform(_c16#299954, lambdafunction(lambda element#300457, lambda element#300457, false)) AS _c16#300459, transform(_c17#299955, lambdafunction(lambda element#300460, lambda element#300460, false)) AS _c17#300462, transform(_c18#299956, lambdafunction(lambda element#300463, lambda element#300463, false)) AS _c18#300465, transform(_c19#299957, lambdafunction(lambda element#300466L, lambda element#300466L, false)) AS _c19#300468, transform(_c20#299958, lambdafunction(lambda element#300469, lambda element#300469, false)) AS _c20#300471, transform(_c21#299959, lambdafunction(lambda element#300472, lambda element#300472, false)) AS _c21#300474, transform(_c22#299960, lambdafunction(lambda element#300475, lambda element#300475, false)) AS _c22#300477, transform(_c23#299961, lambdafunction(lambda element#300478, lambda element#300478, false)) AS _c23#300480, ... 20 more fields]

[2024-06-01T00:46:12.163Z] E               +- Scan ExistingRDD[_c0#299938,_c1#299939,_c2#299940,_c3#299941L,_c4#299942,_c5#299943,_c6#299944,_c7#299945,_c8#299946,_c9#299947,_c10#299948,_c11#299949,_c12#299950,_c13#299951,_c14#299952,_c15#299953,_c16#299954,_c17#299955,_c18#299956,_c19#299957,_c20#299958,_c21#299959,_c22#299960,_c23#299961,... 20 more fields]

[2024-06-01T00:46:12.163Z] 

[2024-06-01T00:46:12.163Z] ../../../spark-3.5.1-bin-hadoop3/python/pyspark/errors/exceptions/captured.py:185: IllegalArgumentException
</pre>
</details>

--- a/integration_tests/src/main/python/hive_parquet_write_test.py
+++ b/integration_tests/src/main/python/hive_parquet_write_test.py
@@ -19,7 +19,7 @@ from conftest import is_databricks_runtime
 from data_gen import *
 from hive_write_test import _restricted_timestamp
 from marks import allow_non_gpu, ignore_order
-from spark_session import with_cpu_session, is_before_spark_320
+from spark_session import with_cpu_session, is_before_spark_320, is_spark_351_or_later
 
 # Disable the meta conversion from Hive write to FrameData write in Spark, to test
 # "GpuInsertIntoHiveTable" for Parquet write.
@@ -55,7 +55,7 @@ _hive_map_gens = [simple_string_to_string_map_gen] + [MapGen(f(nullable=False), 
 _hive_write_gens = [_hive_basic_gens, _hive_struct_gens, _hive_array_gens, _hive_map_gens]
 
 # ProjectExec falls back on databricks due to no GPU version of "MapFromArrays".
-fallback_nodes = ['ProjectExec'] if is_databricks_runtime() else []
+fallback_nodes = ['ProjectExec'] if is_databricks_runtime() or is_spark_351_or_later() else []
 
 
 @allow_non_gpu(*(non_utc_allow + fallback_nodes))

--- a/integration_tests/src/main/python/spark_session.py
+++ b/integration_tests/src/main/python/spark_session.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -219,6 +219,9 @@ def is_spark_341():
 
 def is_spark_350_or_later():
     return spark_version() >= "3.5.0"
+
+def is_spark_351_or_later():
+    return spark_version() >= "3.5.1"
 
 def is_spark_330():
     return spark_version() == "3.3.0"

